redhat-developer/odo-fork
# Kdo push

## What is the purpose of this change? What does it change?

Adds the `push` command. It currently only creates the deployment, service and secret for the component. It does not yet push files to the pod, that will be added in a future PR.

This PR also pulls in further ODO logic related to application, component and labels.

Added comments prefixed with `TODO-KDO:` for areas of focus for the near future.

## How to test changes?

1. Clone the nodejs sample https://github.com/openshift/nodejs-ex
1. Run `kdo create`
1. Run `kdo push`

Current output should look like
```
Validation

 ✓  Validating component [5ms]

 ✓  Checking component [3ms]

Configuration changes
 ✓  Initializing component

 ✓  Creating component [73ms]
 ✓  Changes successfully pushed to component
```

Run `kubectl get all` within the project namespace. The output should contain a pod, service, deployment and replicaset. 

Run `kubectl get secret` and there should be a secret for the component.

Run the automated tests using `make test`
